pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -1216,6 +1225,59 @@ func (p *PravegaCluster) validateConfigMap() error {
 	return nil
 }
 
+// ValidateAuthenticationSettings checks for correct options passed to pravega
+// when authentication is enabled/disabled.
+func (p *PravegaCluster) ValidateAuthenticationSettings() error {
+	if p.Spec.Authentication.Enabled == true {
+		newkey, _ := p.Spec.Pravega.Options["autoScale.controller.connect.security.auth.enable"]
+		oldkey, _ := p.Spec.Pravega.Options["autoScale.authEnabled"]
+
+		if newkey == "" && oldkey == "" {
+			return fmt.Errorf("autoScale.controller.connect.security.auth.enable field is not present")
+		}
+		if newkey == "" {
+			if oldkey != "true" {
+				return fmt.Errorf("autoScale.authEnabled should be set to true")
+			}
+		}
+		if oldkey == "" {
+			if newkey != "true" {
+				return fmt.Errorf("autoScale.controller.connect.security.auth.enable should be set to true")
+			}
+		}
+		if oldkey != "" && newkey != "" {
+			if newkey != "true" || oldkey != "true" {
+				return fmt.Errorf("autoScale.controller.connect.security.auth.enable/autoScale.authEnabled should be set to true")
```

Comment:
Since at this point user is setting both the Options , we can modify the Error message as:
`Both autoScale.controller.connect.security.auth.enable and autoScale.authEnabled should be set to true`